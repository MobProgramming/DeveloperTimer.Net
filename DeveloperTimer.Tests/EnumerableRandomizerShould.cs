using System.Collections.Generic;
using ApprovalTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace DeveloperTimer.Tests
{
    [TestClass]
    public class EnumerableRandomizerShould
    {
        [TestMethod]
        public void ShouldRandomizeEnumerable()
        {
            GetRandomNumber randomNumberGenerator = max => max >= 2 ? 2 : max;

            IEnumerable<string> values = new[] { "one", "two", "three", "four", };

            var randomizer = new EnumerableRandomizer(randomNumberGenerator);

            Approvals.VerifyAll(randomizer.Randomize(values), "Result");
        } 
    }
}