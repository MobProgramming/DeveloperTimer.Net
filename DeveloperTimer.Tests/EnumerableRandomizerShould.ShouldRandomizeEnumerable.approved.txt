Result[0] = three
Result[1] = four
Result[2] = two
Result[3] = one
